client/app/private-database/configuration/private-database-configuration.controller.js

```diff
@@ -24,41 +25,73 @@ angular.module("App").controller(
 
         getConfigurationDetails () {
             this.loading = true;
-            this.privateDatabaseService
+            return this.privateDatabaseService
                 .getConfigurationDetails(this.productId)
                 .then((config) => {
-                    this.configurations = config.details;
-                    this.baseConfiguration = angular.copy(config.details);
+                    this.configurations = config.details.map((field) => this.convertAvailableValues(field));
                 })
-                .catch((err) => {
-                    this.alerter.alertFromSWS(this.$scope.tr("privateDatabase_configuration_error"), err, this.$scope.alerts);
+                .catch((error) => {
+                    this.alerter.error(this.translator.tr("privateDatabase_configuration_error"), this.$scope.alerts.main);
+                    this.$q.reject(error);
                 })
                 .finally(() => {
                     this.loading = false;
                 });
         }
 
+        convertAvailableValues (field) {
+            field.description = this.getFieldDescriptionTranslated(field);
+            field.availableValues = field.availableValues.map((value) => ({ id: value, text: value }));
+            if (field.key === "autocommit") {
+                PrivateDatabaseConfigurationsCtrl.convertFieldAsToggle(field);
+                field.availableValues = [{ id: "0", text: " OFF" }, { id: "1", text: "ON" }];
```

Comment:
for field ```autocommit``` I want the user to configure the parameter with a toggle. I have confirmed with CX that ON/OFF is a good value for every language. Id is never display to user and only used for the API.

Reply:
I don't really understand why you need to declare an ID for this. A simple array of string is enought no ?

Otherwise, the ID field in a `string` instead of `number` ;D